Sources/web3swift/EthereumAPICalls/Ethereum/Eth+GetTransactionDetails.swift

```diff
@@ -8,9 +8,11 @@ import BigInt
 import Core
 
 extension Web3.Eth {
-    public func transactionDetails(_ txhash: Data) async throws -> TransactionDetails {
-        guard let hexString = String(data: txhash, encoding: .utf8)?.addHexPrefix() else { throw Web3Error.dataError }
-        let requestCall: APIRequest = .getTransactionByHash(hexString)
-        return try await APIRequest.sendRequest(with: self.provider, for: requestCall).result
+    public func transactionDetails(_ txHash: Data) async throws -> TransactionDetails {
+        try await transactionDetails(txHash.toHexString().addHexPrefix())
+    }
```

Comment:
As I understand the issue this is the only required change to fix it. I remember that I've struggled with this bit of code, and actually just brute force methods to convert Data to hexstring to make test green. 

But also I'm insists to not expand API to passing `String` argument in addition to `Data` argument — we should not move to the way of flexibility at the expense of consistency. Means that Node call expects that it will get data (as a hexString, but let's leave it on Vitalik's) so we inside the lib should also treat it this way.

Reply:
But also, I'm not sure for now, but it might be that I've tried this way to data -> hexstring conversion, but the tests were red this way. But I've just didn't dig here though, so it you know what you're doing, please feel free to change this bit of pipeline.

Reply:
Yes, the main change/fix this PR is focused on is changing `String(data: txhash, encoding: .utf8)` to `txHash.toHexString()`.

-----
The reason I've added the overloaded functions that accept hashes as `String` is the fact that we receive hashes as `String` from API responses, then we convert it to `Data` to call functions like `func transactionDetails`, and inside of this function we convert the hash back again into `String`. We can reduce these 3 steps to 1 step - just use `String`, but...

I'd advocate for using `Data` instead but it's a topic for later discussion. I'll remove overloaded functions, that's easy to add back if we want to.